### PR TITLE
[1.8.9] Various net.minecraft.world.chunk mappings

### PR DESCRIPTION
**Pull request by Reselim**
Still incomplete, but a little bit less now.

Most of these have been reverse engineered from scratch, however some mappings were taken from 1.17.x yarn.

### CODE REVIEW

**Review by BoogieMonster1O1 — APPROVED** [review]

**Comment by BoogieMonster1O1** [comment]
Is this ready for merge?

File: mappings/net/minecraft/entity/MobSpawnerHelper.mapping [hunk 1]
```diff
@@ -1,5 +1,16 @@
 CLASS net/minecraft/class_107 net/minecraft/entity/MobSpawnerHelper
+	METHOD method_498 findSpawnLocation (Lnet/minecraft/class_99;II)Lnet/minecraft/class_1372;
+		ARG 0 world
+		ARG 1 chunkX
+		ARG 2 chunkY
```
Comment (Grayray75) [inline mappings/net/minecraft/entity/MobSpawnerHelper.mapping:5]: here

File: mappings/net/minecraft/world/World.mapping [hunk 2]
```diff
@@ -370,6 +374,9 @@ CLASS net/minecraft/class_99 net/minecraft/world/World
 	METHOD method_374 tick ()V
 	METHOD method_376 setAmbientDarkness (I)V
 		ARG 1 ambientDarkness
+	METHOD method_377 isSpawnChunk (II)Z
+		ARG 1 chunkX
+		ARG 2 chunkY
```
Comment (Grayray75) [inline mappings/net/minecraft/world/World.mapping:379]: here

File: mappings/net/minecraft/world/chunk/ServerChunkProvider.mapping [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	METHOD <init> (Lnet/minecraft/class_1635;Lnet/minecraft/class_404;Lnet/minecraft/class_396;)V
+		ARG 1 world
+		ARG 2 chunkWriter
+		ARG 3 chunkProvider
+	METHOD method_6025 getChunks ()Ljava/util/List;
+	METHOD method_6026 saveEntities (Lnet/minecraft/class_399;)V
+		ARG 1 chunk
+	METHOD method_6028 (II)V
+		ARG 1 chunkX
+		ARG 2 chunkY
```
Comment (Grayray75) [inline mappings/net/minecraft/world/chunk/ServerChunkProvider.mapping:19]: and here

File: mappings/net/minecraft/world/chunk/ChunkProvider.mapping [hunk 4]
```diff
@@ -6,11 +6,31 @@ CLASS net/minecraft/class_396 net/minecraft/world/chunk/ChunkProvider
...
+		ARG 3 z
+	METHOD method_1324 (Lnet/minecraft/class_396;Lnet/minecraft/class_399;II)Z
+		ARG 1 chunkProvider
+		ARG 2 chunk
+		ARG 3 x
+		ARG 4 z
+	METHOD method_1325 handleInitialLoad (Lnet/minecraft/class_399;II)V
+		ARG 1 chunk
+		ARG 2 x
+		ARG 3 y
```
Comment (Grayray75) [inline mappings/net/minecraft/world/chunk/ChunkProvider.mapping:21]: y => z

File: mappings/net/minecraft/world/chunk/Chunk.mapping [hunk 5]
```diff
@@ -60,51 +71,87 @@ CLASS net/minecraft/class_399 net/minecraft/world/chunk/Chunk
...
 	METHOD method_1377 setInhabitedTime (J)V
 		ARG 1 inhabitedTime
-	METHOD method_1378 (Lnet/minecraft/class_1372;)I
+	METHOD method_1378 getBlockData (Lnet/minecraft/class_1372;)I
 		ARG 1 pos
 	METHOD method_1379 setChunkLoaded (Z)V
 		ARG 1 loaded
+	METHOD method_1381 (II)V
+		ARG 1 x
+		ARG 2 y
```
Comment (Grayray75) [inline mappings/net/minecraft/world/chunk/Chunk.mapping:114]: y => z

File: mappings/net/minecraft/world/dimension/Dimension.mapping [hunk 6]
```diff
@@ -14,6 +14,9 @@ CLASS net/minecraft/class_411 net/minecraft/world/dimension/Dimension
 		ARG 2 tickDelta
 	METHOD method_1468 getById (I)Lnet/minecraft/class_411;
 		ARG 0 id
+	METHOD method_1469 (II)Z
+		ARG 1 x
+		ARG 2 y
```
Comment (Grayray75) [inline mappings/net/minecraft/world/dimension/Dimension.mapping:19]: y => z

**Review by Grayray75 — CHANGES_REQUESTED** [review]
There are still some chunk y left

**Comment by Reselim** [comment]
done

**Review by arthurbambou — APPROVED** [review]

**Review by Grayray75 — APPROVED** [review]

**Pull request merged**